draft-ietf-quic-transport.md

```diff
@@ -2979,21 +2979,57 @@ packet.
 expectations about what implementations do with packets that have errors after
 valid frames? -->
 
+
+## Generating Acknowledgements {#generating-acks}
+
+Endpoints acknowledge all packets they receive and process. However, only
+ack-eliciting packets (see {{QUIC-RECOVERY}}) trigger the sending of an ACK
+frame.  Packets that are not ack-eliciting are only acknowledged when an ACK
+frame is sent for other reasons.
+
+When sending a packet for any reason, an endpoint should attempt to bundle an
+ACK frame if one has not been sent recently. Doing so helps with timely loss
+detection at the peer.
+
+In general, frequent feedback from a receiver improves loss and congestion
+response, but this has to be balanced against excessive load generated by a
+receiver that sends an ACK frame in response to every ack-eliciting packet.  The
+guidance offered below seeks to strike this balance.
+
 ### Sending ACK Frames
 
-An endpoint sends ACK frames to acknowledge packets it has received and
-processed.
+An ACK frame SHOULD be generated for at least every second ack-eliciting packet.
+This recommendation is in keeping with standard practice for TCP {{?RFC5681}}.
 
-Packets containing only ACK frames are not congestion controlled, so there are
-limits on how frequently they can be sent.  An endpoint MUST NOT send more than
-one ACK-frame-only packet in response to receiving an ACK-eliciting packet
-(one containing frames other than ACK and/or PADDING).  An endpoint MUST NOT
-send a packet containing only an ACK frame in response to a non-ACK-eliciting
-packet (one containing only ACK and/or PADDING frames), even if there are
-packet gaps which precede the received packet. Limiting ACK frames avoids an
-infinite feedback loop of acknowledgements, which could prevent the connection
-from ever becoming idle. However, the endpoint acknowledges non-ACK-eliciting
-packets when it sends an ACK frame.
+A receiver's delayed acknowledgment timer SHOULD NOT exceed the current RTT
+estimate or the value it indicates in the `max_ack_delay` transport parameter.
+This ensures an acknowledgment is sent at least once per RTT when packets
+needing acknowledgement are received.  The sender can use the receiver's
+`max_ack_delay` value in determining timeouts for timer-based retransmission.
```

Comment:
Does this text `SHOULD NOT exceed the current RTT estimate` mean the peer could/should use a min of the max_ack_delay and the current RTT when computing probe timeouts?
```
PTO = smoothed_rtt + max(4*rttvar, kGranularity) + min(max_ack_delay, smoothed_rtt)
```

Reply:
Yes, probably.  This text is pre-existing, and maybe with max_ack_delay the 1 ack per RTT recommendation should be removed, but that's not an editorial change.